### DISCUSSION TITLE
false merged segmentation mask

### DISCUSSION THREAD

**Discussion by UnholyCarrot** [ultralytics/ultralytics#18260] [Q&A]
Hi there,
My YOLOv8-n model for segmentation trained on a custom dataset where there is only one class. Eventhough it was trained on a large data set, it still produces false possitives which are merged masks for two objects. I want to consider them as negative samples and train them to model but cannot find a solution except creating a "non-class" and label them so. My question is that how can i solve such problem?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @UnholyCarrot, thank you for reaching out and sharing your question with the Ultralytics community 🚀! 

We highly recommend starting with the [Docs](https://docs.ultralytics.com) to explore key resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. These resources include insights that might help refine your segmentation task setup. If you're confronting a persistent problem, the docs are often a great place to find potential solutions!

If this is a 🐛 Bug Report, please help us debug it faster by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/), including code, dataset sample examples, and outputs.

If this is a custom training ❓ Question, it is helpful to share additional information such as what preprocessing steps you’ve applied, specifics about your dataset, or any logs of the training process. Additionally, you might find insights in our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), which cover best practices for improving YOLOv8 performance. Since you are dealing with merged masks and considering negative samples, exploring data augmentation strategies might also help enhance your dataset.

You can also join the Ultralytics community to collaborate with others. Head to [Discord](https://discord.com/invite/ultralytics) 🎧 for live discussions, visit [Discourse](https://community.ultralytics.com) for in-depth Q&A, or connect with peers on [Reddit](https://reddit.com/r/Ultralytics) for additional community support.

## Upgrade

Ensure your environment is up-to-date by upgrading to the latest `ultralytics` release along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Simply run the following command in a clean [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

For a smooth experience, you might consider running YOLO in any of the following verified environments equipped with all dependencies like [CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled:
- **Notebooks** with free GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM - see [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI - see [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image** - see [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

When the CI badge is green, it confirms that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing successfully across macOS, Windows, and Ubuntu. These tests include all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/). 

This is an automated response 🤖, but don’t worry—an Ultralytics engineer will review your issue and provide further assistance shortly!

**Comment by glenn-jocher** [comment]
@UnholyCarrot thank you for your question. False positives like merged masks can occur due to overlapping objects or insufficient training diversity. While creating a "non-class" is one approach, you could also improve the dataset by including more examples of challenging or overlapping scenarios. Additionally, fine-tuning the model's hyperparameters or using post-processing techniques like confidence threshold adjustments may help reduce false positives. If the issue persists, consider experimenting with a larger model size or data augmentation strategies to enhance model learning.

**Comment by Y-T-G** [comment]
You can add background images to the dataset by just not providing any labels for those images. They're treated as negative samples.
